Add tests for nested macro_rules edition behavior

ehuss opened the pull request:
This adds tests to check the behavior of how nested macro_rules definitions work across edition boundaries. This covers a change in behavior due to https://github.com/rust-lang/rust/pull/133274.

See https://github.com/rust-lang/rust/issues/135669

jieyouxu commented on tests/ui/editions/nested-macro-rules-edition-local-2024.rs:
```diff
@@ -0,0 +1,20 @@
+//
+// This flips the editions to show the opposite behavior.
+
+//@ aux-crate: nested_macro_rules_dep_2021=nested_macro_rules_dep_2021.rs
+//@ edition:2024
+//@ check-pass
```
Question: could this be coalesced into the `-2021` test to become a test with two differential revisions, with the "root" test crate being revisioned on edition 2021 vs 2024 to more obviously show the difference in behavior?

ehuss replied:
Sounds good. Though I can't think of a way to extend that to the aux-crate, which is only differs in the edition.

jieyouxu replied:
> Though I can't think of a way to extend that to the aux-crate, which is only differs in the edition.

In theory they might be able to `include!` some common file and just differ in the compile-flags, but I don't forsee us changing these two aux files often anyways, so seems overkill. The current duplication seems completely fine.

ehuss committed:
Add tests for nested macro_rules edition behavior

This adds tests to check the behavior of how nested macro_rules
definitions work across edition boundaries. This covers a change in
behavior due to https://github.com/rust-lang/rust/pull/133274.

See https://github.com/rust-lang/rust/issues/135669

jieyouxu reviewed (approved):
Thanks, you can r=me after PR CI is green.

jieyouxu commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 4636dd9347573e81056f88189ac52ee1e8ccc2f7 has been approved by `jieyouxu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=jieyouxu 4636dd9347573e81056f88189ac52ee1e8ccc2f7 -->
<!-- homu: {"type":"Approved","sha":"4636dd9347573e81056f88189ac52ee1e8ccc2f7","approver":"jieyouxu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
